Sprint 08

qudryash opened the pull request:
Спринт 8. Сдача работы на ревью

Выполнено:

- проверен код перед сдачей на ревью;
- исправлена индикация лайков на четных и нечетных изображениях;
- также проверено:

- минимально поддерживаемая версия iOS 13.0;
- в настройках основного таргета отключена сборка на iPad;
- установлена галочка поддержки портретной ориентации (в секции Device Orientation), а в Info удалены соответствующие значения для Supported interface orientations (iPhone) (осталось только "Portrait (bottom home button)" - согласно урока;
- структура файлов совпадает со структурой темы «Инициализация второго проекта»;
 - для проекта подключён Git;

- свёрстан Launch Screen приложения:
	- экран полностью соответствует макету;

- свёрстан главный экран с лентой изображений со следующими параметрами:
	- лента содержит 20 фотографий;
	- между фото есть отступ;
	- края фотографий скруглены;
	- на фото добавлен лейбл с текущей датой;
	- на фото добавлена кнопка лайка;
	- лайк проставлен у ячеек с чётным индексом и не проставлен — у нечётным;
	- размеры, параметры и отступы элементов соответствуют макетам;
	- высота ячейки зависит от размеров фото и является динамической.

- приложение запускается.
- элементы интерфейса адаптируются под разрешения экранов iPhone (вёрстка под iPad не предусмотрена).

qudryash committed:
Спринт 08.
Добавлено:
- минимально поддерживаемая версия iOS 13.0;
- в настройках основного таргета отключена сборка на iPad;
- проставлена только галочка поддержки портретной ориентации в секции Device Orientation, а в Info удалены соответствующие значения для "Supported interface orientations (iPhone)";
- структура файлов совпадает со структурой из урока;
- для проекта подключён Git — в меню Source Control в Xcode вам будут доступны меню управления репозиторием.

qudryash committed:
Спринт 8.
Добавлено:
- экспортирован элемент ("Vector") из Figma в PNG и добавлен в Assets;
- добавлены фирменные цвета из Figma в Assets;
- добавлен логотип на экран LaunchScreen;
- выставлены констрейнты по горизонтали и вертикали;
- обновлен задний фон согласно макета Figma.

qudryash committed:
Спринт 8. Выполнен перенос экрана заставки из Сториборда в LaunchScreen:
- перенесен логотип на экран LaunchScreen;
- выставлены констрейнты по горизонтали и вертикали;
- обновлен задний фон согласно макета Figma;
- логотип, констренты и фон удален из сториборда

qudryash committed:
Спринт 8. Выполнено:
- Добавлена таблица UITableView
- Выполнено переименование ViewController на ImagesListViewController
- выполнена связка с таблицы с контроллером ImagesListViewController
- Выполнено переименование таблицы на ImageList
- Выставлены констрейты для таблицы
- Установили цвет для View и таблицы
- Убрали разделение между ячейками

qudryash committed:
Спринт 8. Добавлено:
- объявлена связь таблицы с dataSource и delegate
- добавлено расширение для ImagesListViewController: UITableViewDelegate и UITableViewDataSource
- добавлены методы в расширениях

qudryash committed:
Спринт 8. Выполнено:
- Создана кастомная ячейка
- Добавлен идентификатор ячейки
- Выполнена верстка ячейки
- Добавлена картинка, кнопка лайка и дата

qudryash committed:
Спринт 8. Выполнено:
- добавлены 20 мок изображений
- выполнена верстка кодом, согласно дизайна
- сконфигурирована ячейка,
- добавлены 20 ячеек согласно мок изображений
- выполнен счет четных и нечетных картинок
- добавлено форматирование даты

qudryash committed:
Спринт 8. Выполнено (см. ниже):

- проверен код перед сдачей на ревью;
- исправлена индикация лайков на четных и нечетных изображениях;
- также проверено:

- минимально поддерживаемая версия iOS 13.0;
- в настройках основного таргета отключена сборка на iPad;
- установлена галочка поддержки портретной ориентации (в секции Device Orientation), а в Info удалены соответствующие значения для Supported interface orientations (iPhone) (осталось только "Portrait (bottom home button)" - согласно урока;
- структура файлов совпадает со структурой темы «Инициализация второго проекта»;
 - для проекта подключён Git;

- свёрстан Launch Screen приложения:
	- экран полностью соответствует макету;

- свёрстан главный экран с лентой изображений со следующими параметрами:
	- лента содержит 20 фотографий;
	- между фото есть отступ;
	- края фотографий скруглены;
	- на фото добавлен лейбл с текущей датой;
	- на фото добавлена кнопка лайка;
	- лайк проставлен у ячеек с чётным индексом и не проставлен — у нечётным;
	- размеры, параметры и отступы элементов соответствуют макетам;
	- высота ячейки зависит от размеров фото и является динамической.

- приложение запускается.
- элементы интерфейса адаптируются под разрешения экранов iPhone (вёрстка под iPad не предусмотрена).

amardanov commented on ImageFeed/IImagesList/ImagesListCell.swift:
```diff
@@ -0,0 +1,18 @@
+    
+    static let reuseIdentifier = "ImagesListCell"
+    
+    @IBOutlet weak var cellImage: UIImageView!
```
Давай сделаем все `@IBOutlet` приватными, а их конфигурацию вынесем в публичную функцию "configure". Так мы скрываем детали реализации ячейки и делаем более удобную конфигурацию ячейки с помощью функции, в которую удобнее будет добавить аргументы, при необходимости

Выглядеть это в ячейке будет примерно так:

```
    @IBOutlet private var cellImage: UIImageView!
    @IBOutlet private var likeButton: UIButton!
    @IBOutlet private var dateLabel: UILabel!

    func configure(isLiked: Bool, date: String, image: UIImage) {
        likeButton.setImage(isLiked ? UIImage(named: "Active Like") : UIImage(named: "No Active Like"), for: .normal)
        dateLabel.text = date
        cellImage.image = image
    }
```

А в контроллере, в функции `configCell()` вот так:
```
        cell.configure(
            isLiked: indexPath.row % 2 == 0,
            date: dateFormatter.string(from: Date()),
            image: image
        )
```

qudryash replied:
Выполнено ✅

amardanov commented on ImageFeed/IImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,80 @@
+            return UITableViewCell()
+        }
+        
+        configCell (for: imageListCell, with: indexPath)
```
лишний пробел перед скобкой остался

qudryash replied:
Убрал

qudryash committed:
Спринт 8. Выполнено:
- устранено замечание ревьюера: Изменен iOS deployment Target с 17.0 на 13.0

qudryash committed:
Спринт 8. Выполнен:
- Аутлеты сделаны приватными в классе ImagesListCell
- Удалены строки кода, отвечающие за реализацию вывода изображения, показ текущей даты, процесс вывода лайков на четных и нечетных картинках)
- Добавлен метод в классе ImagesListCell для отработки значения лайков, вывода картинки и даты)
- Добавлено объявление метода cell.configure из класса ImagesListCell в класс ImagesListViewController
- Удалены лишние пробелы перед скобкой в configCell (for: imageListCell, with: indexPath), теперь: configCell(for: imageListCell, with: indexPath)
- Отлючено выделение ячейки
